### PR TITLE
Support multiple locales. JB#54830 OMP#JOLLA-222

### PR DESCRIPTION
**Pull request by llewelld**
Includes two additional locales which will be chosen based on the OS
localisation settings.

The localisation files currently contiain dummy content. The only
difference is the dnsNotFound.title DTD entity. Switching OS locale to
one of the three supported values and opening an unresolvable domain
should bring up a page showing the correct locale.

### CODE REVIEW

**Comment by llewelld** [comment]
I've updated the PR to use the ESR60 content for the three localse (en-US, fi, ru) taken from the mozilla localization repositories. This is still really just to check things work as I did copied the relevant files manually in this case. We'll need a way to automate it across all supported locales.

**Comment by llewelld** [comment]
This is now ready for review. It may be easier to understand by considering the individual commits rather than everything at once: the second commit adds a tool, the third and fourth add Finnish and Russian localisations using the tool.

File: jsscripts/embedhelper.js [hunk 1]
```diff
@@ -22,12 +22,20 @@ XPCOMUtils.defineLazyServiceGetter(Services, "embedlite",
...
+                                    "@mozilla.org/intl/localeservice;1",
+                                    "mozILocaleService");
+
 var globalObject = null;
 var gScreenWidth = 0;
 var gScreenHeight = 0;
 
 const kEmbedStateActive = 0x00000001; // :active pseudoclass for elements
 
+const availablLocales = [
```
Comment (chriadam) [inline jsscripts/embedhelper.js:35]: typo, available

File: jsscripts/embedhelper.js [hunk 2]
```diff
@@ -63,6 +71,9 @@ EmbedHelper.prototype = {
     addMessageListener("Gesture:ContextMenuSynth", this);
     addMessageListener("embed:ContextMenuCreate", this);
     Services.obs.addObserver(this, "embedlite-before-first-paint", true);
+
+    Logger.debug("Available locales: " + availablLocales.join(", "));
```
Comment (chriadam) [inline jsscripts/embedhelper.js:77]: typo repeated here

File: jsscripts/embedhelper.js [hunk 3]
```diff
@@ -22,12 +22,22 @@ XPCOMUtils.defineLazyServiceGetter(Services, "embedlite",
...
+                                    "@mozilla.org/intl/localeservice;1",
+                                    "mozILocaleService");
+
 var globalObject = null;
 var gScreenWidth = 0;
 var gScreenHeight = 0;
 
 const kEmbedStateActive = 0x00000001; // :active pseudoclass for elements
 
+const availablLocales = [
```
Comment (chriadam) [inline jsscripts/embedhelper.js:35]: typo: available

File: tools/mine-localisation-files.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python3
+#
```
Comment (chriadam) [inline tools/mine-localisation-files.py:2]: license information?

File: tools/mine-localisation-files.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    if retcode == 0:
+        print('Git root: {}'.format(git_root))
+
+        cwd_orig = os.getcwd()
+
+        #os.chdir('/tmp')
+
+        # Clone the repository
+        print('Cloning repository {} to {}'.format(repo_remote, repo_local))
+        retcode = subprocess.call(['hg', 'clone', repo_remote, repo_local, '-U'])
```
Comment (chriadam) [inline tools/mine-localisation-files.py:112]: maybe add a comment to the script header which mentions that mercurial needs to be installed by the developer prior to running the script

File: tools/mine-localisation-files.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+            pass
+
+        # Search for the right files
+        # Copy the files
+        # Transform the files
+
+        print('Duplicating non-localised files')
+
+        # overrides/en-US/brand.dtd
+        shutil. copyfile(git_root + '/overrides/en-US/brand.dtd', folder + '/brand.dtd')
```
Comment (chriadam) [inline tools/mine-localisation-files.py:155]: whitespace after shutil. ?

File: tools/mine-localisation-files.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+        # overrides/en-US/brand.properties
+        shutil. copyfile(git_root + '/overrides/en-US/brand.properties', folder + '/brand.properties')
```
Comment (chriadam) [inline tools/mine-localisation-files.py:158]: also here

File: tools/mine-localisation-files.py (hunk 4 above)
Comment (llewelld) [reply tools/mine-localisation-files.py:2]: Thanks, yes, this should have been there. I've added it in.

File: tools/mine-localisation-files.py (hunk 5 above)
Comment (llewelld) [reply tools/mine-localisation-files.py:121]: That's a good idea. I've added a comment, but also added in a simple check that will warn the user if either git or mercurial aren't installed.

File: jsscripts/embedhelper.js (hunk 1 above)
Comment (llewelld) [reply jsscripts/embedhelper.js]: Thanks for spotting this. Now fixed.

File: jsscripts/embedhelper.js (hunk 2 above)
Comment (llewelld) [reply jsscripts/embedhelper.js]: Fixed here too.

File: jsscripts/embedhelper.js (hunk 3 above)
Comment (llewelld) [reply jsscripts/embedhelper.js]: And here.

File: tools/mine-localisation-files.py (hunk 6 above)
Comment (llewelld) [reply tools/mine-localisation-files.py]: I'm a bit surprised python wasn't more unhappy about this. Thanks for spotting it, I've removed the space.

File: tools/mine-localisation-files.py (hunk 7 above)
Comment (llewelld) [reply tools/mine-localisation-files.py]: Removed throughout. Thanks again.

**Comment by chriadam** [comment]
LGTM, but review from someone else who is more familiar with embedlite-components would be good.

**Comment by llewelld** [comment]
Updated to allow `mine-localisation-files.py` to run with Python 3.6 (anything earlier will cause problems).

**Review by chriadam — APPROVED** [review]

**Review by okodron — APPROVED** [review]

**Comment by llewelld** [comment]
The changes in this PR were incorporated into #9, which has now been merged, so I'm closing this as all of these changes have now been merged into master.

**Pull request closed without merging**